code/obj/item/plants.dm

```diff
@@ -437,15 +437,32 @@
 	desc = "By any other name, would smell just as sweet. This one likes to be called "
 	icon_state = "rose"
 	var/thorned = 1
-	var/list/names = list("Emma", "Olivia", "Ava", "Isabella", "Sophia", "Charlotte", "Mia", "Amelia",
-	"Harper", "Evelyn", "Abigail", "Emily", "Elizabeth", "Mila", "Dakota", "Avery",
-	"Sofia", "Camila", "Aria", "Scarlett", "Liam", "Noah", "William", "James",
-	"Oliver", "Benjamin", "Elijah", "Lucas", "Mason", "Logan", "Alexander", "Ethan",
-	"Jacob", "Michael", "Daniel", "Henry", "Jackson", "Sebastian", "Aiden", "Matthew")
 
 	New()
 		..()
-		desc = desc + pick(names) + "."
+		var/list/possible_names = list()
+		for(var/mob/living/carbon/human/H in mobs)
+			if(isnpcmonkey(H))
+				continue
+			if(iswizard(H))
+				continue
+			if(isnukeop(H))
+				continue
+			if(!H.mind)
+				continue
+			possible_names += H
+		for(var/mob/living/silicon/robot/R in mobs)
```

Comment:
this will probably include AI shells, so roses will like to be called "AI shell #82344"

Reply:
Hmm I didn't think of that. Unfortunate for our borgo friends, but I guess I'll remove it for now.